src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -62,6 +63,21 @@ class GitVersionPlugin implements Plugin<Project> {
         }
     }
 
+    @Memoized
+    private String gitBranchName(Project project) {
+        Git git = gitRepo(project)
+        try {
+            def ref = git.repository.getRef(git.repository.branch)
+            if (ref == null) {
+                return null
+            } else {
+                return ref.getName().substring(Constants.R_HEADS.length())
+            }
+        } catch (Throwable t) {
+            return UNSPECIFIED_VERSION
```

Comment:
hmm, no I don't (just copied + pasted it from other methods). Looking a bit closer at it, it will only be hit in weird exception cases (like a malformed git repo).. Removed this try/catch.